addons/web_editor/static/src/js/widgets.js

```diff
@@ -775,85 +775,83 @@ var fontIconsDialog = Widget.extend({
 });
 
 
-function createVideoNode(url, options) {
+function _createVideoNode(url, options) {
     options = options || {};
 
     // video url patterns(youtube, instagram, vimeo, dailymotion, youku)
     var ytRegExp = /^(?:(?:https?:)?\/\/)?(?:www\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})(?:\S+)?$/;
     var ytMatch = url.match(ytRegExp);
 
-    var igRegExp = /\/\/instagram.com\/p\/(.[a-zA-Z0-9]*)/;
-    var igMatch = url.match(igRegExp);
+    var insRegExp = /(.*)instagram.com\/p\/(.[a-zA-Z0-9]*)/;
+    var insMatch = url.match(insRegExp);
 
-    var vRegExp = /\/\/vine.co\/v\/(.[a-zA-Z0-9]*)/;
-    var vMatch = url.match(vRegExp);
+    var vinRegExp = /\/\/vine.co\/v\/(.[a-zA-Z0-9]*)/;
+    var vinMatch = url.match(vinRegExp);
 
     var vimRegExp = /\/\/(player.)?vimeo.com\/([a-z]*\/)*([0-9]{6,11})[?]?.*/;
     var vimMatch = url.match(vimRegExp);
 
-    var dmRegExp = /.+dailymotion.com\/(video|hub)\/([^_]+)[^#]*(#video=([^_&]+))?/;
+    var dmRegExp = /.+dailymotion.com\/(video|hub|embed)\/([^_]+)[^#]*(#video=([^_&]+))?/;
     var dmMatch = url.match(dmRegExp);
 
-    var youkuRegExp = /\/\/v\.youku\.com\/v_show\/id_(\w+)\.html/;
-    var youkuMatch = url.match(youkuRegExp);
+    var ykuRegExp = /(.*).youku\.com\/(v_show\/id_|embed\/)(.+)/;
+    var ykuMatch = url.match(ykuRegExp);
+
+    var $video = $('<iframe>').width(1280).height(720).attr('frameborder', 0).addClass('o_video_dialog_iframe');
+    var $alert = $('<div/>').addClass('alert o_video_dialog_iframe');
+    var video_type = 'yt';
+
+    if(!/^(http:\/\/|https:\/\/|\/\/)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/i.test(url)){
+        $alert.addClass('alert-danger').text(_t('The provided url it\'s not valid' ));
+        return {$content: $alert, success: false};
+    }
+
+    var autoplay = options.autoplay ? "?autoplay=1" : "?autoplay=0";
```

Comment:
Use `'` instead of `"` for technical strings (not a fan but this is the convention we chose).